e2etest/bgptests/bgp.go

```diff
@@ -1108,11 +1113,17 @@ var _ = ginkgo.Describe("BGP", func() {
 					if err == nil {
 						return fmt.Errorf("route not injected from %s", c.Name)
 					}
+					err = checkRouteInjected(speakerPods, pairingIPFamily, toFilter, c.RouterConfig.VRF)
```

Comment:
not in the scope of this PR but I'd change the name to isRouteInjected and the return value to be a boolean. Based on that you can return an error in the closure. It's very misleading the way it is (return error when route is injected).

Reply:
No you are right, I am fixing it. This started simpler but didn't evolve in a readable way.